### PR TITLE
Update Docker Build Instructions and Docker Compose

### PR DESCRIPTION
**Pull request by tylerwhardy**
This pull request represents the following changes:
Updates documentation to reflect build tested to work on Ubuntu 24.04 x86 (Removed references but kept compliant syntax in Docker Compose)
- Migrates build to Docker build (Deprecated this out of PR)
- Migrates Docker Compose v1 application to Docker Compose v2 plugin (KEPT)
- Changes docker-compose.yml syntax to be v2 complaint (KEPT)
- Adds nginx proxy option to Docker-compose file to present Crypt on the network (Deprecated this out of PR)

Incorporates changes proposed on https://github.com/grahamgilbert/Crypt-Server/pull/131 to fix 1 vulnerability by pinning transitive dependencies:

https://snyk.io/vuln/SNYK-PYTHON-SELENIUM-6062316

### CODE REVIEW

**Comment by arubdesu** [comment]
If selenium testing is no longer being leveraged it can be dumped, btw... (I'm to `git blame` for that one)

File: docker-compose.yml [hunk 1]
```diff
@@ -25,3 +24,15 @@ services:
       - ${PWD}/crypt.db:/home/app/crypt/crypt.db # This will do a local database. For production you should use postgresql
       - ${PWD}/fvserver/settings.py:/home/app/crypt/fvserver/settings.py # Load in your own settings file
     restart: always
+    
+  nginx:
```
Comment (grahamgilbert) [inline docker-compose.yml:28]: Why are we adding this?

File: docs/Docker.md [hunk 2]
```diff
@@ -1,25 +1,82 @@
...
+ Debug Mode: false
+ Plugins:
+  buildx: Docker Buildx (Docker Inc.)
+    Version:  0.12.1
+    Path:     /usr/libexec/docker/cli-plugins/docker-buildx
+```
+
+Build the Dockerfile:
+``` bash
+docker buildx build -t crypt-server .
```
Comment (grahamgilbert) [inline docs/Docker.md:50]: Is this necessary for most people? I would prefer the docs to reference the built image.

File: docs/Docker.md [hunk 3]
```diff
@@ -1,25 +1,82 @@
 # Using Docker
 
+## Server Initialization
+This was last tested on Ubuntu 24.04 x86
+Note: For this document, macadmins/crypt-server and crypt-server are used interchangeably. macadmins/crypt-server references the downloadedy option and crypt-server represents the local build option.
```
Comment (grahamgilbert) [inline docs/Docker.md:5]: Stick to the built image please.

**Review by grahamgilbert — CHANGES_REQUESTED** [review]

File: docker-compose.yml (hunk 1 above)
Comment (tylerwhardy) [reply docker-compose.yml]: Hi. I saw a mention of using the nginx proxy for convenience and put it in. I removed it. I did keep the Docker compose file in the v2 standard.

File: docs/Docker.md (hunk 3 above)
Comment (tylerwhardy) [reply docs/Docker.md:5]: Will revise to indicate using the built image. 

Is it possible to get the image rebuilt more frequently to incorporate upstream library changes?

File: docs/Docker.md (hunk 2 above)
Comment (tylerwhardy) [reply docs/Docker.md:50]: Will remove

File: docs/Docker.md (hunk 3 above)
Comment (grahamgilbert) [reply docs/Docker.md]: The latest image is built after every commit to the master branch. That of course does not stop you from building your own.

**Review by grahamgilbert — APPROVED** [review]

**Pull request merged**